iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/CopyTreeImplTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.impl;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.docker.parser.grammar.DockerLexicalGrammar;
+import org.sonar.iac.docker.parser.utils.Assertions;
+import org.sonar.iac.docker.tree.api.CopyTree;
+import org.sonar.iac.docker.tree.api.DockerTree;
+import org.sonar.iac.docker.tree.api.ParamTree;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.iac.common.testing.TextRangeAssert.assertTextRange;
+import static org.sonar.iac.docker.tree.impl.DockerTestUtils.parse;
+
+class CopyTreeImplTest {
+  @Test
+  void matchingSimple() {
+    Assertions.assertThat(DockerLexicalGrammar.COPY)
+      .matches("COPY src dest")
+      .matches("copy src dest")
+      .matches("COPY dest")
```

Comment:
What does this mean for the Docker image? src == dest?

Reply:
For Docker image this will not "compile" and will throw an error.
I discussed with Pierre-Yves about how to parse this, the way that SSLR is implemented prevent from handling such use case : `COPY (word)+ word`
It won't work because at first it will match everything as **src** `(word)+` but then there is nothing left for **dest** `word`, SSLR won't backtrack/try to match **src** differently.

I choose to accept such use case, but I guess we could also throw a parse error or something similar, open to challenge.

Reply:
Okay fine.